### PR TITLE
Add service status metrics

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java [hunk 1]
```diff
@@ -117,7 +127,32 @@ private void updateNodeMetrics(Node node) {
...
+            numberOfServices = 0;
+        } else {
+            Map<ServiceStatus, Long> servicesCount = services.stream().collect(
+                    Collectors.groupingBy(ServiceInstance::serviceStatus, Collectors.counting()));
+
+            numberOfServices = servicesCount.values().stream().mapToLong(Long::longValue).sum();
+
+            metric.set(
+                    "numberOfServicesUp",
+                    servicesCount.getOrDefault(ServiceStatus.UP, 0L), context);
```
Comment (bjorncs) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java]: Consider having each argument to `metric.set()` on separate lines or all on same lines. Now it is easy to mistake `context` as an argument to `getOrDefault`.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/ServiceMonitorStub.java [hunk 2]
```diff
@@ -89,4 +90,8 @@ private ServiceStatus getHostStatus(String hostname) {
         return status;
     }
 
+    @Override
+    public ServiceModel getServiceModelSnapshot() {
+        throw new IllegalStateException("getServicemodelSnapshot has not been implemented");
```
Comment (bjorncs) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/ServiceMonitorStub.java]: Consider using `UnsupportedOperationException`.

**Review by bjorncs — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java]: Thanks, that's usually what I do. Fixed.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/ServiceMonitorStub.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/ServiceMonitorStub.java]: Thanks, for some reason I unable to find a more appropriately named class. Fixed.

**Pull request merged**